infrastructure/010-aws-api/modules/dynamodb/dynamodb.tf

```diff
@@ -0,0 +1,53 @@
+module "this" {
+  source = "terraform-aws-modules/dynamodb-table/aws"
+
+  name      = "${var.name_prefix}-${random_id.this.hex}"
+  hash_key  = "PartitionKey"
+  range_key = "SortKey"
+
+  # Enable streaming to allow for global replication
+  stream_enabled   = var.is_global
+  stream_view_type = var.is_global ? "NEW_AND_OLD_IMAGES" : null
+
+  server_side_encryption_enabled     = true
+  server_side_encryption_kms_key_arn = var.server_side_encryption_kms_key_arn
+
+  attributes = [
+    {
+      name = "PartitionKey"
+      type = "S"
+    },
+    {
+      name = "SortKey"
+      type = "S"
+    },
+    {
+      name = "Value"
+      type = "S"
+    }
+  ]
+
+  global_secondary_indexes = [
+    {
+      hash_key        = "SortKey"
+      name            = "InverseIndex"
+      projection_type = "KEYS_ONLY"
+      range_key       = "PartitionKey"
+    },
+    {
+      hash_key        = "Value"
+      name            = "ValueIndex"
+      projection_type = "ALL"
+      range_key       = "PartitionKey"
+    }
```

Comment:
This is a weird one. I don't expect to use value often, but for value-inverted lookups, we can populate a special key that allows for reverse lookup by value. We use the whole projection under the understanding that this will be used in special cases only and any key included will need to be sent.